lazy_dataset/core.py

```diff
@@ -1022,21 +1022,16 @@ def tile(self, reps: int, shuffle: bool = False) -> 'Dataset':
             >>> ds
                 ListDataset(len=5)
               MapDataset(_pickle.loads)
-                ListDataset(len=5)
-              MapDataset(_pickle.loads)
-                ListDataset(len=5)
-              MapDataset(_pickle.loads)
-            ConcatenateDataset()
+            TileDataset(repetitions=3)
             >>> list(ds)
             [1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5]
         """
-        datasets = [self] * reps
         if shuffle:
-            datasets = [
-                ds.shuffle()
-                for ds in datasets
-            ]
-        return self.__class__.concatenate(*datasets)
+            datasets = [self] * reps
+            datasets = [ds.shuffle() for ds in datasets]
+            return self.__class__.concatenate(*datasets)
+        else:
+            return TileDataset(self, reps)
```

Comment:
To minimize overhead: Could your return self, when reps is one? concatenate does this already.